### ISSUE TITLE
Single threaded executor hangs forever

### ISSUE THREAD

**Issue by akousari** [jenetics/jenetics#111]
When I configure an executor with a single thread the execution never finishes. Only by introducing a second thread the final result gets displayed. 

This code would hang indefinitely. Change int threads = 2; and the result gets displayed
```

import java.util.concurrent.Executors;

import org.jenetics.DoubleChromosome;
import org.jenetics.DoubleGene;
import org.jenetics.Genotype;
import org.jenetics.engine.Codec;
import org.jenetics.engine.Engine;
import org.jenetics.engine.EvolutionResult;
import org.jenetics.util.DoubleRange;
import org.jenetics.util.IntRange;
import org.jenetics.util.LongRange;

/**
 * @author <a href="mailto:franz.wilhelmstoetter@gmx.at">Franz WilhelmstÃ¶tter</a>
 * @version 3.4
 * @since 3.3
 */
public class CodecExample {

	// The domain class
	final static class Tuple {
		final int _1;
		final long _2;
		final double _3;

		Tuple(final int v1, final long v2, final double v3) {
			_1 = v1;
			_2 = v2;
			_3 = v3;
		}

		@Override
		public String toString() {
			return String.format("_1: %d - _2: %d - _3: %f", _1, _2, _3);
		}
	}

	// The fitness function. No need to know anything about GAs. Decoupling of
	// the problem function from the GA usage.
	static double f(final Tuple param) {
	    System.out.println(param._1);
		return param._1 + param._2 + param._3;
	}

	// The encoding/decoding of the problem domain is defined at ONE place.
	static Codec<Tuple, DoubleGene> codec(
		final IntRange v1Domain,
		final LongRange v2Domain,
		final DoubleRange v3Domain
	) {
		return Codec.of(
			Genotype.of(
				DoubleChromosome.of(DoubleRange.of(v1Domain.getMin(), v1Domain.getMax())),
				DoubleChromosome.of(DoubleRange.of(v2Domain.getMin(), v2Domain.getMax())),
				DoubleChromosome.of(v3Domain)
			),

			gt -> new Tuple(
			    
				gt.getChromosome(0).getGene().intValue(),
				gt.getChromosome(1).getGene().longValue(),
				gt.getChromosome(2).getGene().doubleValue()
			)
		);
	}
	

	public static void main(final String[] args) {
        run();
	}

    private static void run() {
        int threads = 1; //Change to 2 to show the result
        //int threads = 2;
        
		// The domain of your fitness function.
		final IntRange domain1 = IntRange.of(0, 100);
		final LongRange domain2 = LongRange.of(0, 1_000_000_000_000L);
		final DoubleRange domain3 = DoubleRange.of(0, 1);

		// The problem domain encoder/decoder.
		final Codec<Tuple, DoubleGene> codec = codec(domain1, domain2, domain3);

		final Engine<DoubleGene, Double> engine = Engine
			.builder(CodecExample::f, codec).executor(Executors.newFixedThreadPool(threads))
			.build();

		final Genotype<DoubleGene> gt = engine.stream()
			.limit(1000)
			.collect(EvolutionResult.toBestGenotype());

		final Tuple param = codec.decoder().apply(gt);
		System.out.println(String.format("Result: \t%s", param));
    }

}


```

**Comment by jenetics** [comment]
I could reproduce this with the `TravelingSalesman` example.
[TravelingSalesman.jstack.txt](https://github.com/jenetics/jenetics/files/270917/TravelingSalesman.jstack.txt)

The error is in the `Engine` class, where I try to calculate the fitness values in parallel. This works for more than one thread, but fails with one. I'm not sure when I have time for the fix. As a workaround, you can execute all operations in the _main_ thread:
```java
final Engine<DoubleGene, Double> engine = Engine
    .builder(CodecExample::f, codec)
    .executor(Runnable::run)
    .build();
```

**Comment by akousari** [comment]
Thanks for looking. In my case I want to use multiple single threaded executors on different problems, so the main thread workaround doesn't work for me unfortunately. I will try to see if I can fix it if I find some time.

**Comment by jenetics** [comment]
Merged into bug fix branch r.3.5.1.
